File: blue_first.rb
class BlueFirst
  attr_reader :award

  def initialize(award)
    @award = award
  end

  def update
    award.expires_in -= 1

    award.quality += 1
    award.quality += 1 if award.expires_in <= 0
    return award.quality = 50 if award.quality >= 50
  end
end

File: blue_first.py
from award import Award

class BlueFirst(Award):
	def __init__(self, expires_in=None, quality=None):
		super(BlueFirst, self).__init__('Blue First', expires_in, quality)

	def update_quality(self):
		if self.quality < 50:
			self.quality += 1

		self.expires_in -= 1
		if self.expires_in < 0 and self.quality < 50:
			self.quality += 1